engine/docker/docker.go

```diff
@@ -427,3 +424,71 @@ func (e *Docker) createNetworkWithRetries(ctx context.Context,
 	}
 	return err
 }
+
+func (e *Docker) pullDockerImage(ctx context.Context, step *spec.Step, pullOpts types.ImagePullOptions, output io.Writer) error {
+	pulled := false
+
+	// automatically pull the latest version of the image if requested
+	// by the process configuration, or if the image is :latest
+	if step.Pull == spec.PullAlways ||
+		(step.Pull == spec.PullDefault && image.IsLatest(step.Image)) {
+		pullerr := e.pullImageWithRetries(ctx, step.Image, pullOpts, output)
+		if pullerr != nil {
+			return pullerr
+		}
+		pulled = true
+	}
+
+	if path, ok := step.Envs[pathEnvVariable]; ok {
+		// pull the image if not already pulled
+		if !pulled {
+			// check if already present on VM else pull
+			pulled = true
+			if exists := e.imageExistsLocally(ctx, step.Image); !exists {
```

Comment:
This is not correct. Just add in ci-manager run script this line at the top (according to os arch)
```
export PATH=$PATH:<Path in env>
```

Reply:
remove path env from run step

Reply:
I tried above but that didn't work out.

> remove path env from run step
@shubham149 We set the envs from github action step in runner (https://github.com/drone-runners/drone-runner-aws/blob/ac86ad01ee1ef42e2af6f8065e35cd40ab0714a5/command/harness/step.go#L57). I believe we need to set this for all the subsequent steps.

Reply:
Path that is exported here https://github.com/drone-runners/drone-runner-aws/blob/ac86ad01ee1ef42e2af6f8065e35cd40ab0714a5/command/harness/step.go#L57 won't be useful for steps with docker container.

Reply:
@shubham149 I see, so would it make sense to remove this logic for container steps at all ? I was earlier thinking to do this but wasn't sure how it can impact the execution.